juju/juju
# Remove the fanconfigurer worker

In the context of dropping the fan networking support on 4.0+, this patch removes the fanconfigurer worker, which applied the fan configuration using `fanctl`, which is part of the ubuntu-fan package and no longer installed after the patch https://github.com/juju/juju/pull/17410 lands.

All api endpoints regarding fanconfigurer have also been removed.

## Checklist



## QA steps

Bootstrap on aws and deploy:

```
juju bootstrap aws c
juju add-model m
juju deploy ubuntu --to lxd
```
and make sure there are no errors on the logs.

Also, 3.6->4.0 migration should not be broken:
```
juju_36 bootstrap aws 36
juju_36 add-model m
juju_36 deploy ubuntu --to lxd
# install this patch at this point
juju bootstrap aws c40
juju switch c36:admin/m
juju migrate c40
```
Check the logs and the status after migration, the model should have been correctly migrated.
## Links

**Jira card:** JUJU-6048
